exercises/sam_jenkins/ex01_basics/src/mapFunctions.cpp

```diff
@@ -0,0 +1,49 @@
+#include <unordered_map>
+#include <sstream>
+#include <vector>
+#include "stringManip.h"
+
+
+
+using std::string; using std::unordered_map; using std::vector; using std::pair;
+
+unordered_map<string, int> countof(const string& input)
+{
+	//create a unordered map and string input stream
+	unordered_map<string, int>wordtable;
+	std::istringstream iss(input);
+	//create an iterator to loop through the string
+	auto i = input.begin();
+	while (i != input.end()) {
+		//find the first non space character
+		i = find_if(i, input.end(), not_space);
```

Comment:
I believe you can use `find_if_not(i, input.end(), space);` here instead of `find_if` with the `not_space`. This will save you having to define `not_space`.